proposals/20220926-contrib-component-guidelines.md

```diff
@@ -0,0 +1,55 @@
+# Guidelines for /contrib Components
+
+**Authors**: Kimonas Sotirchos kimwnasptd@arrikto.com
+
+The motivation behind this proposal is to fully document expectations and
+requirements that componetns under `/contrib` should satisfy. This will make it
+more clear how to use a component, how it integrates with Kubeflow, the problems
+it tries to solve as well as dependency versions.
+
+## Goals
+
+* Document requirements that components under `/contrib` should satisfy
+* Introduce a process for deprecating unmaintained components
+
+## Non-Goals
+
+* Get into the discussion of which components are considered "Kubeflow" components
+    * The assumption until now is that components under the Kubeflow GitHub Org live
+      under the `/apps` dir, and all others under `/contrib`
+* Provide a migration plan for components to move out from `/contrib`
+* Modify the [`example/kustomization.yaml`](https://github.com/kubeflow/manifests/blob/master/example/kustomization.yaml) with new components
+
+## Proposal
+
+### Component Requirements
+
+Components living under `/contrib` should satisfy some strict requirements to
+ensure they are always usable by end-users and contain complementary documentation.
+
+These are the following items that all components under `/contrib` should have:
+1. A `README.md` file that documents:
+    * The K8s version and KF component versions that the component needs
+    * Instructions on how someone can install the component in a Kubeflow cluster
+        * Since Kubeflow manifests have standardized on [Kustomize](https://kustomize.io/)
+          we expect all manifests to be a kustomize packages
+    * How to use the component as part of Kubeflow (examples)
+    * The problems it tries to solve and the value it brings
+2. An OWNERS file with at least 2 users
+
+At this point we don't want to provide too much of a strict structure for the
+README. Developers are free to expose any other information in the README that
+they find fit, as long as the above info is exposed.
+
+
+### Deprecation plan
+
+The proposed criteria for deciding that a component should be deprecated are:
+1. The component can not be installed in the minimum K8s version supported by Kubeflow
+2. The manifests are not working as expected and result in undeployable Pods
+3. The documented examples do not work as expected
```

Comment:
Hey @kimwnasptd, I think this is a great initiative to clarify the `/contrib` directory! 

I was wondering, who will be testing these components to make sure that these requirements are met each release? Will this be driven by feedback from end users? or is there a plan to have some tests around this?

Reply:
For now I'd say that it's driven by user feedback.

Although we can also have some simple GH Actions that test that at least the manifests of the components could be applied in a cluster. But I think we can have this discussion after we do a first cleanup and figure out which components satisfy the initial requirements

Reply:
I would propose for at least the new elements that we ask the sponsoring project to help in the test plan, test execution and documentation.   For example KServe, Seldon, BentoML, Infinstor, and Tecton could help with KServe, Seledon, Bento, Concurrent and Feast respectively.

Reply:
I've addressed the testing requirements below with a distinct commit. I'll mark this one as resolved for now, to keep the effort going and be able to merge a first iteration. 

If something feels way too off though feel free to re-open this discussion